propkatraj/tests/test_regression.py

```diff
@@ -52,46 +52,19 @@ def pka_from_file(filename):
     return resnums, pkas
 
 
-@pytest.mark.parametrize('selection', ['protein', 'array', 'list'])
+@pytest.mark.parametrize('selection', ['default', 'protein'])
 def test_single_frame_regression(tmpdir, u, selection):
-    """Single frame propkatraj call compared against same frame written by
-    MDA
-    """
-    with tmpdir.as_cwd():
-        if selection == 'array':
-            selection = u.select_atoms('protein').ix
-        elif selection == 'list':
-            selection = u.select_atoms('protein').ix.tolist()
-
-        pkas = propkatraj.get_propka(u, sel=selection, stop=1)
-
-        # load reference data
-        ref_resnums, ref_pkas = pka_from_file(PSF_FRAME_ZERO_PKA)
-
-        # test residue numbers
-        resnums = pkas.columns.to_numpy()
-        assert_equal(resnums, ref_resnums)
-
-        # test pka values
-        assert_almost_equal(pkas.values[0], ref_pkas, decimal=2)
-
-
-@pytest.mark.parametrize('selection', ['protein', 'ag'])
-def test_single_frame_regression_analysisbase(tmpdir, u, selection):
     """Single frame PropkaTraj regression test"""
     with tmpdir.as_cwd():
-        if selection == 'ag':
-            pkatraj = propkatraj.PropkaTraj(u.select_atoms('protein'),
+        if selection != 'default':
+            pkatraj = propkatraj.PropkaTraj(u.select_atoms(selection),
                                             select=None)
-        elif selection == 'array':
-            selection = u.select_atoms('protein').ix
-            pkatraj = propkatraj.PropkaTraj(u, select=selection)
         else:
             pkatraj = propkatraj.PropkaTraj(u)
 
         pkatraj.run(stop=1)
 
-        # load reference data
+        # load reference data, makes testing arbitrary selections difficult
```

Comment:
True, you'd probably have to slice the reference data based on the `selection.residues`. Even then, you might get differences because propka might be missing interactions. Definitely leave the updated comment!